### PR TITLE
PLT-6203 : Add Unclaimed Payouts in ContractState Payload

### PR DESCRIPTION
**Pull request by nhenin**
- **JIRA** : [PLT-6203](https://input-output.atlassian.net/browse/PLT-6203?atlOrigin=eyJpIjoiZjRjOTE4ZmQzMDE2NDQyM2FhYTMxNzA1MzdkYjUxNTMiLCJwIjoiaiJ9)

- **Runtime REST** 
- **Value** : Add Unclaimed Payouts in the endpoint `Get /Contract/:contractId/` 
- Addition in Payload : `Get /Contract/:contractId/`

```json
{
  "links": {
    "transactions": "string"
  },
  "resource": {
    "block": {
      "blockHeaderHash": "string",
      "blockNo": 0,
      "slotNo": 0
    },
    "continuations": "string",
    "contractId": "98d601c9307dd43307cf68a03aad0086d4e07a789b66919ccf9f7f7676577eb7#1",
    "metadata": {},
    "roleTokenMintingPolicyId": "string",
    "state": {
      "accounts": [
        [
          null
        ]
      ],
      "boundValues": [
        [
          null
        ]
      ],
      "choices": [
        [
          null
        ]
      ],
      "minTime": 0
    },
    "status": "unsigned",
    "tags": {},
    "txBody": {
      "cborHex": "string",
      "description": "string",
      "type": "string"
    },
    "unclaimedPayouts": [
      {
        "payoutId": "98d601c9307dd43307cf68a03aad0086d4e07a789b66919ccf9f7f7676577eb7#1",
        "role": "string"
      }
    ],
    "utxo": "98d601c9307dd43307cf68a03aad0086d4e07a789b66919ccf9f7f7676577eb7#1",
    "version": "v1"
  }
}
```

**Done** 

**Todo** 

[PLT-6203]: https://input-output.atlassian.net/browse/PLT-6203?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Contract/State.hs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Data.Text (Text)
+import Data.Word (Word64)
+import GHC.Generics (Generic)
+import qualified Language.Marlowe.Core.V1.Semantics.Types as Semantics
+import Language.Marlowe.Runtime.Web.Orphans ()
+
+import Language.Marlowe.Runtime.Web.Types
+  (BlockHeader, MarloweVersion, Metadata, PolicyId, TextEnvelope, TxOutRef, TxStatus)
+
+data ContractState = ContractState
```
Comment (jhbertra) [inline marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Contract/State.hs:19]: Can you please move this back to `Types.hs`? I don't want to have this many types modules. I think we could split it out by `Contracts` `Transactions` and `Withdrawals` for example, but this is too granular. And if we do that, I'd want to refactor everything so it's all consistent.

File: marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Contract/State.hs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  , tags :: Map Text Metadata
+  , metadata :: Map Word64 Metadata
+  , status :: TxStatus
+  , block :: Maybe BlockHeader
+  , initialContract :: Semantics.Contract
+  , currentContract :: Maybe Semantics.Contract
+  , state :: Maybe Semantics.State
+  , utxo :: Maybe TxOutRef
+  , txBody :: Maybe TextEnvelope
+  , unclaimedPayouts :: [Payout]
```
Comment (jhbertra) [inline marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Contract/State.hs:33]: Here we could reuse the existing `PayoutRef` type, which also includes the `assets`

Comment (jhbertra) [reply marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Contract/State.hs:33]: Oh snap, no it doesn't!

Comment (jhbertra) [reply marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Contract/State.hs:33]: Everything is missing the `assets`... I've gotta fix that!

File: marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Types.hs [hunk 3]
```diff
@@ -231,8 +232,15 @@ data ContractState = ContractState
   , state :: Maybe Semantics.State
   , utxo :: Maybe TxOutRef
   , txBody :: Maybe TextEnvelope
+  , unclaimedPayouts :: [Payout]
   } deriving (Show, Eq, Generic)
 
+data Payout
+    = Payout
+        { payoutId :: TxOutRef
+        , role :: Text} -- TODO (N.H) : replace it by an asset (policyId,tokenName) vs just a role (~TokenName)
```
Comment (jhbertra) [inline marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Types.hs:241]: Actually this `role` type is fine; what we want to do is add an `assets` field which says what funds were paid out.

**Review by jhbertra — APPROVED** [review]
Tahnks!!

File: marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Types.hs (hunk 3 above)
Comment (nhenin) [reply marlowe-runtime-web/src/Language/Marlowe/Runtime/Web/Types.hs:241]: ok ok

File: marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/DTO.hs [hunk 4]
```diff
@@ -52,6 +52,7 @@ import Data.Coerce (coerce)
 import Data.Data (Typeable)
 import Data.List.NonEmpty (NonEmpty(..))
 import Data.Map (Map)
+import qualified Data.Map as M
```
Comment (yveshauser) [inline marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/DTO.hs:55]: cosmetics: the line below there is already `Map`

**Review by yveshauser — APPROVED** [review]
Thanks, looks good!

**Pull request merged**